source/mbed_trace.c

```diff
@@ -259,6 +259,10 @@ static void mbed_trace_default_print(const char *str)
 }
 void mbed_tracef(uint8_t dlevel, const char *grp, const char *fmt, ...)
 {
+    if (NULL == m_trace.line) {
+        return;
```

Comment:
Hmm, that will not work. If m_trace is NULL, mbed_trace_reset_tmp() will Hard fault. I would still keep the original version.

This is to prevent mbed-client-linux-example from hard faulting if it happens that user has old version of the git tree and all yotta-dependencies have been updated to mbed-trace. This would mean that the user initializes the library with wrong trace library and tries to use unintialized pointers.